platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/crypto/EnhancedKeyStoreLoader.java

```diff
@@ -0,0 +1,1094 @@
+     *                              specified {@code entry}.
+     * @throws NullPointerException if {@code entry} is {@code null}.
+     */
+    @NonNull
+    private PublicKey extractPublicKeyEntity(@NonNull final Object entry) throws KeyLoadingException {
+        Objects.requireNonNull(entry, MSG_ENTRY_NON_NULL);
+
+        try {
+            final JcaPEMKeyConverter converter = new JcaPEMKeyConverter();
+            final PEMDecryptorProvider decrypter = new JcePEMDecryptorProviderBuilder().build(keyStorePassphrase);
+
+            return switch (entry) {
+                case SubjectPublicKeyInfo spki -> converter.getPublicKey(spki);
+                case PEMKeyPair kp -> converter.getPublicKey(kp.getPublicKeyInfo());
+                case PEMEncryptedKeyPair ekp -> converter.getPublicKey(
+                        ekp.decryptKeyPair(decrypter).getPublicKeyInfo());
+                default -> throw new KeyLoadingException("Unsupported entry type [ entryType = %s ]"
+                        .formatted(entry.getClass().getName()));
+            };
+        } catch (IOException e) {
+            throw new KeyLoadingException(
+                    "Unable to extract a public key from the specified entry [ entryType = %s ]"
+                            .formatted(entry.getClass().getName()),
+                    e);
+        }
+    }
+
+    /**
+     * Helper method used by {@link #extractEntityOfType(Object, Class)} for extracting a {@link PrivateKey} from the
+     * specified {@code entry}.
+     *
+     * @param entry the entry loaded from the store.
+     * @return the {@link PrivateKey} extracted from the specified {@code entry}.
+     * @throws KeyLoadingException  if an error occurred while attempting to extract the {@link PrivateKey} from the
+     *                              specified {@code entry}.
+     * @throws NullPointerException if {@code entry} is {@code null}.
+     */
+    @NonNull
+    private PrivateKey extractPrivateKeyEntity(@NonNull final Object entry) throws KeyLoadingException {
+        Objects.requireNonNull(entry, MSG_ENTRY_NON_NULL);
+
+        try {
+            final JcaPEMKeyConverter converter = new JcaPEMKeyConverter();
+            final PEMDecryptorProvider decrypter = new JcePEMDecryptorProviderBuilder().build(keyStorePassphrase);
+            final InputDecryptorProvider inputDecrypter =
+                    new JceInputDecryptorProviderBuilder().build(new String(keyStorePassphrase).getBytes());
+
+            return switch (entry) {
+                case PrivateKeyInfo pki -> converter.getPrivateKey(pki);
+                case PKCS8EncryptedPrivateKeyInfo epki -> converter.getPrivateKey(
+                        epki.decryptPrivateKeyInfo(inputDecrypter));
+                case PEMKeyPair kp -> converter.getPrivateKey(kp.getPrivateKeyInfo());
+                case PEMEncryptedKeyPair ekp -> converter.getPrivateKey(
+                        ekp.decryptKeyPair(decrypter).getPrivateKeyInfo());
```

Comment:
It looks good that this function can extract different formats. 

However, since `keyStorePassphrase` cannot be `null`, essentially the loader only supports `PEMEncryptedKeyPair`. Am I right?

Just wondering if we should avoid supporting other formats.

Reply:
@leninmehedy The loader supports all formats listed. Restricting the array parameter to non-null simply means the are not allowed to pass in a `null` value but they may still pass in a zero length `char` array.